app/frontend/src/components/MapSearch.tsx

```diff
@@ -88,44 +92,55 @@ export default function MapSearch({
     setSearchOptionsLoading(false);
   };
 
-  const searchSubmit = (
-    value: string | SearchOption | null,
-    reason: AutocompleteChangeReason
-  ) => {
-    console.log(reason);
+  const searchSubmit = (value: string, reason: AutocompleteChangeReason) => {
     if (reason === "blur") {
       setOpen(false);
       return;
     }
-    if (!(value as SearchOption)?.name) {
-      setCity(value as string);
+    const searchOption = searchOptions.find((o) => value === o.name);
+    if (!searchOption) {
+      setValue(value);
       //create-option is when enter is pressed on user-entered string
       if (reason === "create-option") {
-        loadSearchOptions(value as string);
+        loadSearchOptions(value);
       }
     } else {
-      const searchOption = value as SearchOption;
       setMarker(searchOption.location);
       setOpen(false);
     }
   };
 
   return (
-    <Autocomplete
-      label="My location"
-      //hint="Press enter to search the map, then customize the text."
-      size="small"
-      options={searchOptions}
-      getOptionLabel={(o) => (!!o.name ? o.name : String(o))}
-      loading={searchOptionsLoading}
-      open={open}
-      onBlur={() => setOpen(false)}
-      onChange={(_, value, reason) => searchSubmit(value, reason)}
-      onInputChange={(_, value) => setCity(value)}
-      freeSolo
-      multiple={false}
-      disableClearable={false}
-      className={classes.root}
-    />
+    <Box className={classes.root}>
+      <Autocomplete
```

Comment:
I was getting console warnings that this autocomplete is switching between controlled and non-controlled, but not sure why.